sdk/identity/Azure.Identity/src/Credentials/AzurePipelinesCredential.cs

```diff
@@ -86,21 +88,19 @@ internal async ValueTask<AccessToken> GetTokenCoreAsync(bool async, TokenRequest
             }
         }
 
-        internal HttpMessage CreateOidcRequestMessage(string serviceConnectionId, AzurePipelinesCredentialOptions options)
+        internal HttpMessage CreateOidcRequestMessage(AzurePipelinesCredentialOptions options)
         {
-            string CollectionUri = options.CollectionUri ?? throw new CredentialUnavailableException("AzurePipelinesCredential is not available: environment variable SYSTEM_TEAMFOUNDATIONCOLLECTIONURI is not set.");
-            string projectId = options.TeamProjectId ?? throw new CredentialUnavailableException("AzurePipelinesCredential is not available: environment variable SYSTEM_TEAMPROJECTID is not set.");
-            string planId = options.PlanId ?? throw new CredentialUnavailableException("AzurePipelinesCredential is not available: environment variable SYSTEM_PLANID is not set.");
-            string jobId = options.JobId ?? throw new CredentialUnavailableException("AzurePipelinesCredential is not available: environment variable SYSTEM_JOBID is not set.");
-            string systemToken = options.SystemAccessToken ?? throw new CredentialUnavailableException("AzurePipelinesCredential is not available: environment variable SYSTEM_ACCESSTOKEN is not set.");
-            string hubName = options.HubName ?? throw new CredentialUnavailableException("AzurePipelineCredential is not available: environment variable SYSTEM_HOSTTYPE is not set.");
+            string serviceConnectionId = ServiceConnectionId ?? throw new CredentialUnavailableException("AzurePipelinesCredential is not available: Either serviceConnectionId was not supplied to the credential or the environment variable AZURESUBSCRIPTION_SERVICE_CONNECTION_ID is not set.");
```

Comment:
even though this is a super self-explanatory exception message, can we embed an aka.ms link to a new TSG section for it?  So if we do get some sort of unusual customer reported issues we can update the TSG section and customers hopefully we'll have a better debugging experience.

Reply:
Good callout. I was planning to wait till we had the TSG text and then retro fit it here.